### PR TITLE
Revert the revert of RHOAM RBAC

### PR DESCRIPTION
**Pull request by alanmoran**
This pr reverts the revert (#568) of #562 

The plan is for this PR to be 👍 again by owners and for it to be merged by APAC functional lead (@dofinn)  on Monday morning (APAC) to allow CS SRE to test. 

 @mciccone is the owner of this PR.

### CODE REVIEW

File: hack/00-osd-managed-cluster-config-integration.yaml.tmpl [hunk 1]
```diff
@@ -1467,7 +1467,7 @@ objects:
       managed.openshift.io/gitHash: ${IMAGE_TAG}
       managed.openshift.io/gitRepoName: ${REPO_NAME}
       managed.openshift.io/osd: 'true'
-    name: layered-sre-authorization-addon-rhmi-operator
+    name: layered-sre-authorization-rhmi-addon-rhmi-operator
```
Comment (alanmoran) [inline hack/00-osd-managed-cluster-config-integration.yaml.tmpl:1470]: Review how resourceApplyMode works.

Comment (alanmoran) [reply hack/00-osd-managed-cluster-config-integration.yaml.tmpl:1470]: Specify "Sync" to allow deleting existing objects that were previously in the resources list.

**Comment by alanmoran** [comment]
/assign @lisa 

could you please review and approve if happy? Please do not merge as @dofinn will do this on Monday morning (APAC)

**Comment by lisa** [comment]
@dofinn feel free to remove my hold when you merge.

/lgtm

**Pull request merged**